### DISCUSSION TITLE
Support sqlite backend with rke2

### DISCUSSION THREAD

**Discussion by retr0h** [rancher/rke2#4626] [General]
### Background

- We use k3s as a single node appliance, primarily as an edge device
- We are switching off k3s to rke2 primarily for the FIPS features

We ran thorough performance  testing, and with much tuning, found the system to meet our needs.  *However*, in migrating to rke2, we have found etcd is failing and causing rke2 to no longer function (see below).  We understand etcd has [hardware recommendations](https://etcd.io/docs/v3.4/op-guide/hardware/), which we don't currently meet.  Which is ultimately the request.  

```
Aug 11 15:26:40 notice k3s: {"level":"info","ts":"2023-08-11T15:26:40.610Z","logger":"client","caller":"v3@v3.5.7-k3s1/client.go:210","msg":"Auto sync endpoints failed.","error":"context deadline exceeded"}
Aug 11 15:26:49 notice k3s: {"level":"warn","ts":"2023-08-11T15:26:49.913Z","logger":"etcd-client","caller":"v3@v3.5.7-k3s1/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc001d86000/127.0.0.1:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = context deadline exceeded"}
```

### Ask

Could we have an option to run **rke2** with **sqlite**, similar to k3s.  This would solve many of our ongoing issues.

**Comment by brandond** [comment]
We do not currently plan to support any datastores other than etcd. RKE2 intentionally does not target lightweight edge devices or hosts without suitable low-latency storage; these are targeted by k3s.

**Reply by davidgaster** [reply]
I would like to highlight the benefits of RKE2, particularly its built-in FIPS compliance, which is a pivotal requirement for government systems. RKE2's adherence to security technical implementation guides (STIGs) out of the box is a significant advantage.

If we were to consider K3s as an alternative, we would face the challenge of implementing equivalent STIGs from the ground up, right? Without that, etcd's suitability for embedded or single-node devices, such as laptops that regularly transition between networks, is pretty limited.

**Reply by brandond** [reply]
Please don't consider a year old discussion post indicative of the current state. See https://github.com/rancher/rke2/issues/5957 and other follow-up work. This is still experimental at the moment, but is possible.

**Reply by davidgaster** [reply]
Was not aware, and there wasn't an update on this issue. This is great news! Will give this a try, thank you!

**Reply by brandond** [reply]
This isn't an issue, it's a discussion, and we don't usually take the time to circle back on those unless someone comments.

**Reply by davidgaster** [reply]
No need to split hairs. I figured as much, which is why I commented. Great to hear this is in development.
